src/ImageSharp/Formats/Png/PngEncoder.cs

```diff
@@ -72,13 +72,11 @@ public void Encode<TPixel>(Image<TPixel> image, Stream stream)
         /// <param name="image">The <see cref="Image{TPixel}"/> to encode from.</param>
         /// <param name="stream">The <see cref="Stream"/> to encode the image data to.</param>
         /// <returns>A <see cref="Task"/> representing the asynchronous operation.</returns>
-        public async Task EncodeAsync<TPixel>(Image<TPixel> image, Stream stream)
+        public Task EncodeAsync<TPixel>(Image<TPixel> image, Stream stream)
             where TPixel : unmanaged, IPixel<TPixel>
         {
-            using (var encoder = new PngEncoderCore(image.GetMemoryAllocator(), image.GetConfiguration(), new PngEncoderOptions(this)))
-            {
-                await encoder.EncodeAsync(image, stream).ConfigureAwait(false);
-            }
+            using var encoder = new PngEncoderCore(image.GetMemoryAllocator(), image.GetConfiguration(), new PngEncoderOptions(this));
+            return encoder.EncodeAsync(image, stream);
```

Comment:
I'm sure this one needs awaiting otherwise the encoder will be disposed before the `Task` is awaited/activated and things will possibly exploded, but only if we end up having to perform a `CopyToAsync` which is unlikely to be happening the unit test

Reply:
I’m cool with switching it back.